pkg/oidc/verifier.go

```diff
@@ -32,6 +32,12 @@ type ClaimsSignature interface {
 	SetSignatureAlgorithm(algorithm jose.SignatureAlgorithm)
 }
 
+type IDClaims interface {
+	Claims
+	GetSignatureAlgorithm() jose.SignatureAlgorithm
```

Comment:
i think we should either move the `GetSignatureAlgorithm` to the `Claims` interface or implement it on the `IDTokenClaims` struct, because currently the `TokenClaims` implements it, but not the whole `IDClaims` interface, which feels weird:
![image](https://user-images.githubusercontent.com/9405495/224248995-71583ff0-97ec-4880-858f-8417d9372ca9.png)

Reply:
The problem is with `JWTTokenRequest` which does not implement this method.
By setting the method on `Claims`,  In `VerifyJWTAssertion` the following functions will complain:

- `oidc.CheckAudience`
- `oidc.CheckExpiration`
- `oidc.CheckIssuedAt`

`JWTTokenRequest` does have a `SetSignatureAlgorithm` method which is a no-op. I could add `GetSignatureAlgorithm`, but it would always return an empty string. I felt that would be a dirtier solution with the risk that future devs will use `JWTTokenRequest` in places where the difference does matter.

Reply:
https://github.com/zitadel/oidc/pull/283#discussion_r1132295959

Reply:
![image](https://user-images.githubusercontent.com/5411563/224313684-d40b936b-9844-4950-a741-266640dda711.png)

Reply:
Misread your comment, yes I could implement it for `IDTokenClaims` instead, but it already extends the `TokenClaims` type, so it's already implemented by extension. Adding one extra method doesn't hurt tho. Let me know what you want to do.